server/py/framework/api/utils.py

```diff
@@ -204,11 +206,87 @@ def _generate_function_and_task_from_submit_run_body(db_session: Session, data):
 
 
 async def submit_run(
-    db_session: Session, auth_info: mlrun.common.schemas.AuthInfo, data
+    db_session: Session,
+    auth_info: mlrun.common.schemas.AuthInfo,
+    background_tasks: fastapi.BackgroundTasks,
+    data,
 ):
-    _, _, _, response = await run_in_threadpool(
-        submit_run_sync, db_session, auth_info, data
+    from services.api.utils.endpoints import (
+        start_model_endpoint_creation_background_task,
     )
+
+    response = None
+
+    try:
+        fn, task = _generate_function_and_task_from_submit_run_body(db_session, data)
+        run_db = get_run_db_instance(db_session)
+        fn.set_db_connection(run_db)
+
+        track_models = getattr(fn.spec, "track_models", False)
+        logger.info(
+            "Starting model endpoint creation?",
+            track_models=track_models,
+            background_tasks=str(background_tasks),
+            db_session=str(db_session),
+        )
+
+        if track_models and background_tasks and db_session:
+            project = task["metadata"]["project"]
+            function_name = fn.metadata.name
+            (
+                fn,
+                model_endpoint_creation_task_name,
+                _,
+            ) = await start_model_endpoint_creation_background_task(
+                project=project,
+                name=function_name,
+                background_tasks=background_tasks,
+                function=fn.to_dict(),
+                db_session=db_session,
+                is_batch=True,
+            )
+            fn = mlrun.new_function(
+                runtime=fn,
+                project=project,
+                name=function_name,
+            )
```

Comment:
Maybe better to first create the function, then start the bg task, just in case the function creation fails for some reason (wrong parameters or whatever).